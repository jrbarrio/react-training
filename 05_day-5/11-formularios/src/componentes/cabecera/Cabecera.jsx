import { useState } from 'react';

export default function Cabecera({ texto }) {
    
    const [ contador, setContador] = useState(0);
    
    return (
        <div>
            <h1 data-testid="h1tag" className="header">{ texto }</h1>
            <p data-testid="pCont">{ contador }</p>
            <button onClick={ () => setContador(contador + 1) }>Incrementa</button>
        </div>
    );
}