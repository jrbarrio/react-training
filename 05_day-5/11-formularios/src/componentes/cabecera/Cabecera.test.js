import { fireEvent, render } from '@testing-library/react';
import Cabecera from './Cabecera';

describe('<Cabecera />', () => {
    

    it('renderiza correctamente', () => {
        const { asFragment } = render(<Cabecera texto="adios"></Cabecera>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Introduce texto en la cabecera', () => {
        const { getByTestId, getByText } = render(<Cabecera texto="hola"></Cabecera>);
        expect(getByTestId('h1tag')).toHaveTextContent("hola");
        expect(getByText('hola')).toHaveClass("header");
    });

    it('Incrementa pulsando boton', () => {
        const { getByText, getByTestId } = render(<Cabecera texto="cabecera2"></Cabecera>)
        fireEvent.click(getByText('Incrementa'))
        expect(getByTestId('pCont')).toHaveTextContent("1");
    });
})