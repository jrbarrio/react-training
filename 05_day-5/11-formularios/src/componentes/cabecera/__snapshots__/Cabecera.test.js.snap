// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cabecera /> renderiza correctamente 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="header"
      data-testid="h1tag"
    >
      adios
    </h1>
    <p
      data-testid="pCont"
    >
      0
    </p>
    <button>
      Incrementa
    </button>
  </div>
</DocumentFragment>
`;
